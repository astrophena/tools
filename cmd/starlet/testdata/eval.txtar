update: start
-- bot.star --
load("@starlet//tg.star", "tg")

def handle(update):
    # Must succeed.
    res = eval("print(greeting)", environ = {
        "greeting": "Hello, world!",
    })
    tg.send_message(update["message"]["chat"]["id"], res)
    # Must ignore non-string environ keys.
    res2 = eval("print(greeting)", environ = {
        "greeting": "Hello, world!",
        123: "Whassup, homie!",
    })
    # Should be an error (invalid arguments).
    res3 = eval()
