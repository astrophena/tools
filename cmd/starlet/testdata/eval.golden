[
  {
    "method": "sendMessage",
    "args": {
      "chat_id": 123456789,
      "entities": [],
      "link_preview_options": {
        "is_disabled": true
      },
      "reply_markup": {},
      "text": "Hello, world!\n\n"
    }
  },
  {
    "method": "sendMessage",
    "args": {
      "chat_id": 123456789,
      "entities": [
        {
          "length": 112,
          "offset": 25,
          "type": "pre"
        }
      ],
      "link_preview_options": {
        "is_disabled": true
      },
      "text": "‚ùå Something went wrong:\n\nTraceback (most recent call last):\n  //bot.star:15:16: in handle\nError in eval: eval: missing argument for code\n\n"
    }
  }
]